support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/web/response/callback/OAuth20AuthorizationCodeAuthorizationResponseBuilder.java

```diff
@@ -50,6 +50,11 @@ public ModelAndView build(final WebContext webContext, final String clientId,
             holder.getResponseType(), holder.getGrantType());
         LOGGER.debug("Generated OAuth code: [{}]", code);
         configurationContext.getTicketRegistry().addTicket(code);
+        val tgt = holder.getTicketGrantingTicket();
+        if (tgt != null) {
```

Comment:
Use `Optional.ofNullable(tgt).ifPresent()`

Reply:
As the `updateTicket` method may throw an exception, it leads to a more complex source code:

```java
        Optional.ofNullable(ticketGrantingTicket).ifPresent(tgt -> {
            LOGGER.debug("Updating parent ticket-granting-ticket [{}]", tgt);
            try {
                configurationContext.getTicketRegistry().updateTicket(tgt);
            } catch (final Exception e) {
                LOGGER.error("Unable to update ticket-granting-ticket [{}]", tgt, e);
            }
        });
```

Are you sure?

Reply:
No, it's simpler than that:

```java
Optional.ofNullable(ticketGrantingTicket).ifPresent(tgt -> {
    FunctionUtils.doAndHandle(() -> configurationContext.getCentralAuthenticationService().updateTicket(tgt),
    						  () -> LOGGER.error("Unable to update ticket-granting-ticket [{}]", tgt, e))
});
```

This is pseudocode, but should work. In general, you should not use the ticket-registry for CRUD ops; it's not transaction-bound.

Reply:
Right, I should not use the ticket registry directly.

The final syntax is the following as a result must always be returned (`public static <R> Supplier<R> doAndHandle(final CheckedSupplier<R> function, final CheckedFunction<Throwable, R> errorHandler)`):

```java
Optional.ofNullable(ticketGrantingTicket).ifPresent(tgt -> {
    FunctionUtils.doAndHandle(() -> configurationContext.getCentralAuthenticationService().updateTicket(tgt),
            throwable -> {
                LOGGER.error("Unable to update ticket-granting-ticket [{}]", tgt, throwable);
                return null;
            });
});
```

In terms of readability, I prefer the original `null` check. But this might be me...

Reply:
You're right. It's probably uglier than it needs to be. Your call. I was just reading that JDK 19 might make this sort of thing a bit easier. But for now, let's stick with what works.

Reply:
Just to be sure: do you want me to revert to the original `null` check or keep the new "more complex" syntax?

Reply:
Probably the null syntax looks better, but it's your call if you want to switch. LGTM to me as is anyway.

Reply:
OK. Will keep things as is. This is ready for merge.